analysis/basestats/analyze.h

```diff
@@ -38,13 +46,81 @@ class Analysis {
+            ranks[req.oid]++;
         }
     }
 
     void outputStats() {
         for(auto & it: counters) {
             std::cout << it.first << " " << it.second << "\n";
         }
+        double sum = 0;
+        double mean = (double)counters["TotalBytes"] / (double)counters["RequestCount"];
+        for(auto & it: reqsizes) {
+            sum += (it - mean) * (it - mean);
+        }
+        double stddev = sqrt(sum / (double)counters["RequestCount"]);
+        double scv = stddev * stddev / (mean * mean);
+        std::cout << "Mean " << mean << "\n";
+        std::cout << "StdDev " << stddev << "\n";
+        std::cout << "SquaredCoeffVar " << scv << "\n";
+
+        double maxTs = counters["MaxTs"];
+        for (auto & it: reqs) {
+            Request curr = it.second[0];
+            if (it.second.size() == 1) {
+                if (curr.ts < maxTs - (double)172800) {
+                    counters["OneHitWondersObjs"]++;
+                    counters["OneHitWondersBytes"]+=curr.size;
+                    if(onehitwonids.count(curr.oid)==0) {
+                        counters["UniqueOneHitWondersObjs"]++;
+                        counters["UniqueOneHitWondersBytes"]+=curr.size;
+                        onehitwonids.insert(curr.oid);
+                    }
+                }
+            } else {
+                for (std::size_t i=1; i<it.second.size(); ++i) {
+                    if (curr.ts < it.second[i].ts - (double)172800) {
+                        counters["OneHitWondersObjs"]++;
+                        counters["OneHitWondersBytes"]+=curr.size;
+                        if(onehitwonids.count(curr.oid)==0) {
+                            counters["UniqueOneHitWondersObjs"]++;
+                            counters["UniqueOneHitWondersBytes"]+=curr.size;
+                            onehitwonids.insert(curr.oid);
+                        }
+                    }
+                    curr = it.second[i];
+                }
+            }
+        }
+        std::cout << "% of One Hit Wonder Objects " << (double)counters["OneHitWondersObjs"] / (double)counters["TotalObjs"] * 100 << "\n";
+        std::cout << "% of Unique One Hit Wonder Objects " << (double)counters["UniqueOneHitWondersObjs"] / (double)counters["UniqueObjs"] * 100 << "\n";
+        std::cout << "% of One Hit Wonder Bytes " << (double)counters["OneHitWondersBytes"] / (double)counters["TotalBytes"] * 100 << "\n";
+        std::cout << "% of Unique One Hit Wonder Bytes " << (double)counters["UniqueOneHitWondersBytes"] / (double)counters["UniqueBytes"] * 100 << "\n";
+        std::cout << "% of Unique One Hit Wonder Requests " << (double)onehitwonids.size() / (double)objids.size() * 100 << "\n";
+        // for(auto & it: reqs) {
+        //     std::cout << it.first << "\n"; // << " " << it.second
+        // }
+        std::cout << "Ranks" << "\n";
+        for(auto & it: ranks) {
+            std::cout << it.first << " " << it.second << "\n";
+        }
```

Comment:
It would be more efficient to aggregate the rank data right here, i.e., compute the CDF: how often does each popularity occur in the trace. The object ids (it.first) are fairly long and suboptimal as intermediary format, as they don't really convey much meaning.